### PR TITLE
Add initial support for SuperCollider

### PR DESCRIPTION
**Pull request by pnelson**
Implements syntax highlighting for SuperCollider. See also #7368.

### CODE REVIEW

File: runtime/queries/supercollider/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,101 @@
+; highlights.scm
+; See this for full list: https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md
```
Comment (the-mikedavis) [inline runtime/queries/supercollider/highlights.scm:2]: Helix uses different capture names than neovim. See https://docs.helix-editor.com/master/themes.html#syntax-highlighting

For example captures like `parameter` need to become `variable.parameter`, `property` becomes `variable.other.member`, etc.

File: runtime/queries/supercollider/indents.scm
Comment (the-mikedavis) [inline runtime/queries/supercollider/indents.scm:1]: Helix also has a custom indentation system that these queries will need to be rewritten for, see https://docs.helix-editor.com/master/guides/indent.html

File: runtime/queries/supercollider/locals.scm
Comment (the-mikedavis) [inline runtime/queries/supercollider/locals.scm:1]: For locals we follow tree-sitter: https://tree-sitter.github.io/tree-sitter/syntax-highlighting#local-variables

so there should only be the captures `local.scope`, `local.definition` and `local.reference`

**Review by the-mikedavis — CHANGES_REQUESTED** [review]

**Comment by pnelson** [comment]
I touched up the `highlights.scm` and it's looking good when I run it on [tutorial.scd](https://raw.githubusercontent.com/igramjp/supercollider/master/tutorial.scd) but really struggled with the other tree-sitter files. Are highlights-only an acceptable breakpoint for this PR?

**Review by the-mikedavis — DISMISSED** [review]

**Comment by the-mikedavis** [comment]
Yep highlights-only is a good start :+1:. The other queries can be contributed as follow-ups if anyone is interested in working on them

**Review by the-mikedavis — DISMISSED** [review]

**Review by pascalkuthe — DISMISSED** [review]

**Comment by pascalkuthe** [comment]
needs a rebase

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**